components/net/filemanager_thread.rs

```diff
@@ -137,7 +138,7 @@ impl FileManager {
         thread::Builder::new()
             .name("transfer memory".to_owned())
             .spawn(move || {
-                store.promote_memory(blob_buf, set_valid, sender, origin);
+                store.promote_memory(id, blob_buf, set_valid, sender, origin);
```

Comment:
I think since this happens in another thread, we can't reliably do the assertion in `test_fetch_blob` now that we're not blocking while this completes. Do we need to remove this `spawn`, or change the test?

Reply:
You're right; I can see how we can have ordering problems with this change. I don't see any good reason to spawn a thread for this short operation, so let's remove it.

Reply:
@jdm if possible can you help me understand what this "promote memory" is for? When we send the `FileManagerThreadMsg::PromoteMemory` message, we won't have any guarantee that it completed (though it sounds like that's exactly what we intend based on the issue description).

In `test_fetch_blob` we just call `file_manager.promote_memory` directly so I think I'll be able to get it to pass by removing this spawn. I'm just curious how this will bubble up.

Thanks!

Reply:
@miller-time When there is a uniquely-owned [Blob](https://w3c.github.io/FileAPI/#blob-section) which contains a string of bytes stored in memory, the promotion operation extracts those bytes from the DOM blob object, sends them to the global file manager thread and stores them in a hashtable indexed by a UUID. It then becomes possible for multiple Blob objects to exist referencing this UUID without having to store multiple copies of the string (which could be very large).

In this case we do not need any guarantee that it completed; any subsequent operations involving the blob will send the new UUID, and they will be ordered after the original message that promoted its contents, so the new UUID will match what was stored in the global file manager thread.

Does that make sense?

Reply:
Yep (~mostly). Thanks for the explanation!